ambry-protocol/src/main/java/com/github/ambry/protocol/FileCopyProtocolGetChunkResponse.java

```diff
@@ -0,0 +1,93 @@
+  private PartitionId partitionId;
+  private String fileName;
+  protected ByteBuf chunkData;
+  private final long startOffset;
+  private final long chunkSizeInBytes;
+  private final boolean isLastChunk;
+  private static final int File_Name_Field_Size_In_Bytes = 4;
+
+  public FileCopyProtocolGetChunkResponse(short versionId, int correlationId, String clientId, ServerErrorCode errorCode,
+      PartitionId partitionId, String fileName, ByteBuf chunkData, long startOffset, long chunkSizeInBytes, boolean isLastChunk) {
+    super(RequestOrResponseType.FileCopyProtocolGetChunkResponse, versionId, correlationId, clientId, errorCode);
+    this.partitionId = partitionId;
+    this.fileName = fileName;
+    this.chunkData = chunkData;
+    this.startOffset = startOffset;
+    this.chunkSizeInBytes = chunkSizeInBytes;
+    this.isLastChunk = isLastChunk;
+  }
+
+  public static FileCopyProtocolGetChunkResponse readFrom(DataInputStream stream, ClusterMap clusterMap) throws IOException {
+    RequestOrResponseType type = RequestOrResponseType.values()[stream.readShort()];
+    if (type != RequestOrResponseType.FileCopyProtocolGetChunkResponse) {
+      throw new IllegalArgumentException("The type of request response is not compatible");
+    }
+    short versionId = stream.readShort();
+    int correlationId = stream.readInt();
+    String clientId = Utils.readIntString(stream);
+    ServerErrorCode errorCode = ServerErrorCode.values()[stream.readShort()];
+
+    if(errorCode != ServerErrorCode.No_Error){
+      return new FileCopyProtocolGetChunkResponse(versionId, correlationId, clientId, errorCode,
+          null, null, null, -1, -1, false);
+    }
+
+    PartitionId partitionId = clusterMap.getPartitionIdFromStream(stream);
+    String fileName = Utils.readIntString(stream);
+    long startOffset = stream.readLong();
+    long sizeInBytes = stream.readLong();
+    boolean isLastChunk = stream.readBoolean();
+    int chunkLengthInBytes = stream.readInt();
+    ByteBuf chunk = Unpooled.buffer(chunkLengthInBytes);
+    for(int i = 0; i < chunkLengthInBytes; i++){
+      chunk.writeByte(stream.readByte());
+    }
+    return new FileCopyProtocolGetChunkResponse(versionId, correlationId, clientId, errorCode, partitionId, fileName, chunk, startOffset, sizeInBytes, isLastChunk);
+  }
+  public long sizeInBytes() {
+    return super.sizeInBytes() + partitionId.getBytes().length + File_Name_Field_Size_In_Bytes + fileName.length() + Long.BYTES + Long.BYTES + 1 + Integer.BYTES + chunkData.readableBytes();
+  }
+  public void prepareBuffer(){
+    super.prepareBuffer();
+    bufferToSend.writeBytes(partitionId.getBytes());
+    Utils.serializeString(bufferToSend, fileName, Charset.defaultCharset());
+    bufferToSend.writeLong(startOffset);
+    bufferToSend.writeLong(chunkSizeInBytes);
+    bufferToSend.writeBoolean(isLastChunk);
+    bufferToSend.writeInt(chunkData.readableBytes());
+    bufferToSend.writeBytes(chunkData);
```

Comment:
again, we don't actually have to copy all the bytes over, we can use a composite bytebuf.  check out here https://github.com/linkedin/ambry/blob/master/ambry-protocol/src/main/java/com/github/ambry/protocol/PutRequest.java#L269